### ISSUE TITLE
openssl speed inconsistent behavior

### ISSUE THREAD

**Issue by beldmit** [openssl/openssl#21248]
`openssl speed -provider legacy -evp md4` works normally, though reporting 

```
00D24E505F7F0000:error:0308010C:digital envelope routines:inner_evp_generic_fetch:unsupported:crypto/evp/evp_fetch.c:373:Global default library context, Algorithm (CTR-DRBG : 0), Properties (<null>)
00D24E505F7F0000:error:12000090:random number generator:rand_new_drbg:unable to fetch drbg:crypto/rand/rand_lib.c:577:
```
which is suboptimal but not really important.

At the same time, the command
`openssl speed -evp md4 -provider legacy`
does not work at all:
```
speed: md4 is an unknown cipher or digest
0032D224A17F0000:error:0308010C:digital envelope routines:inner_evp_generic_fetch:unsupported:crypto/evp/evp_fetch.c:373:Global default library context, Algorithm (md4 : 85), Properties (<null>)
0032D224A17F0000:error:0308010C:digital envelope routines:inner_evp_generic_fetch:unsupported:crypto/evp/evp_fetch.c:373:Global default library context, Algorithm (MD4 : 85), Properties ()
0032D224A17F0000:error:03000086:digital envelope routines:evp_md_init_internal:initialization error:crypto/evp/digest.c:254:
```

**Comment by beldmit** [comment]
Didn't check against master branch so labeling 3.0 only

**Comment by lelik107** [comment]
I tried to test speed using OpenSSL v. 3.1.1 and 3.0.9 both light versions windows x64 and got same errors with this command:
openssl.exe speed -evp (ciphers affected:) cast5-cbc,cast,bf, idea, rc2, rc4, seed.

But this works just fine with v. 1.1.1u. And this is something isn't right.

**Comment by beldmit** [comment]
Yes. In 3.0 you should load legacy provider for those algorithms. 

E.g. `openssl speed -provider legacy -evp rc4` should work, I think

**Comment by lelik107** [comment]
> Yes. In 3.0 you should load legacy provider for those algorithms.
> 
> E.g. `openssl speed -provider legacy -evp rc4` should work, I think

You are right about command the test started but here what I have after it:

DC0D0000:error:0308010C:digital envelope routines:inner_evp_generic_fetch:unsupported:crypto\evp\evp_fetch.c:373:Global default library context, Algorithm (CTR-DRBG : 0), Properties (<null>)
DC0D0000:error:12000090:random number generator:rand_new_drbg:unable to fetch drbg:crypto\rand\rand_lib.c:577:
DC0D0000:error:0308010C:digital envelope routines:inner_evp_generic_fetch:unsupported:crypto\evp\evp_fetch.c:373:Global default library context, Algorithm (CTR-DRBG : 0), Properties (<null>)
DC0D0000:error:12000090:random number generator:rand_new_drbg:unable to fetch drbg:crypto\rand\rand_lib.c:577:
DC0D0000:error:0308010C:digital envelope routines:inner_evp_generic_fetch:unsupported:crypto\evp\evp_fetch.c:373:Global default library context, Algorithm (CTR-DRBG : 0), Properties (<null>)
DC0D0000:error:12000090:random number generator:rand_new_drbg:unable to fetch drbg:crypto\rand\rand_lib.c:577:
DC0D0000:error:0308010C:digital envelope routines:inner_evp_generic_fetch:unsupported:crypto\evp\evp_fetch.c:373:Global default library context, Algorithm (CTR-DRBG : 0), Properties (<null>)
DC0D0000:error:12000090:random number generator:rand_new_drbg:unable to fetch drbg:crypto\rand\rand_lib.c:577:
DC0D0000:error:0308010C:digital envelope routines:inner_evp_generic_fetch:unsupported:crypto\evp\evp_fetch.c:373:Global default library context, Algorithm (CTR-DRBG : 0), Properties (<null>)
DC0D0000:error:12000090:random number generator:rand_new_drbg:unable to fetch drbg:crypto\rand\rand_lib.c:577:
DC0D0000:error:0308010C:digital envelope routines:inner_evp_generic_fetch:unsupported:crypto\evp\evp_fetch.c:373:Global default library context, Algorithm (CTR-DRBG : 0), Properties (<null>)
DC0D0000:error:12000090:random number generator:rand_new_drbg:unable to fetch drbg:crypto\rand\rand_lib.c:577:
DC0D0000:error:0308010C:digital envelope routines:inner_evp_generic_fetch:unsupported:crypto\evp\evp_fetch.c:373:Global default library context, Algorithm (CTR-DRBG : 0), Properties (<null>)
DC0D0000:error:12000090:random number generator:rand_new_drbg:unable to fetch drbg:crypto\rand\rand_lib.c:577:

**Comment by beldmit** [comment]
Yes. It's not an error though it looks nasty.

**Comment by lelik107** [comment]
May be but we don't have test results anyway.

**Comment by beldmit** [comment]
`openssl speed -provider legacy -provider default rc4` should do the trick

**Comment by lelik107** [comment]
Thank you this did the job. Can't you tell me how to use -provider-path option in Windows? Example may be.

**Comment by beldmit** [comment]
Have no Windows and no idea :)

**Comment by lelik107** [comment]
Lucky guy ? :) It can be
`openssl speed -provider-path "C:\Program Files\OpenSSL-Win64\bin" -provider legacy -provider default rc4`
I just forgot quotes :)

**Comment by t8m** [comment]
This is very much debatable whether this is a bug or _just not a nice UI experience_.

**Comment by nhorman** [comment]
marking as inactive, will be closed at the completion of 3.4 dev without further inpu

**Comment by levitte** [comment]
> This is very much debatable whether this is a bug or _just not a nice UI experience_.

That's how I characterise this issue
